src/shared/prosemirror-plugins/image-upload.ts

```diff
@@ -297,19 +307,36 @@ export class ImageUploader extends PluginInterfaceView<
         event.stopPropagation();
     }
 
-    createAcceptedFileTypeString(acceptedTypes: string[] = []): string {
-        if (acceptedTypes.length === 0) {
-            return "";
-        }
-        let acceptedTypesString = "Supported file types: ";
-        acceptedTypes.forEach((s, i) => {
-            if (i === acceptedTypes.length - 1) {
-                acceptedTypesString += `or ${s}.`;
-            } else {
-                acceptedTypesString += `${s}, `;
+    getAcceptedFileTypesString(types: string[]): string {
+        let uploadCaptionString = "";
+        const acceptedTypes = types?.sort() || [];
+        const defaultTypes = ["image/gif", "image/jpeg", "image/png"];
+
+        // If the arrays are different, we modify the caption string
+        if (JSON.stringify(acceptedTypes) !== JSON.stringify(defaultTypes)) {
+            if (acceptedTypes.length > 1) {
+                acceptedTypes[acceptedTypes.length - 1] =
+                    "or " + acceptedTypes[acceptedTypes.length - 1];
             }
-        });
-        return acceptedTypesString;
+            uploadCaptionString = acceptedTypes
+                .join(", ")
+                .replace(/image\//g, "");
```

Comment:
Why even have the `image` bit if we're just removing it any time we're displaying it? In this context it will always be an image.

Reply:
We're removing `image/` when we're _displaying_ it, but not always when we're using it. I figured it made the most sense to stick to the [MIME type](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types) to imply the connection to the file input's `accept` attribute. I'd be fine with inverting this if you think the API would be simpler to instead have the consumer provide a list of file extensions.

Let me know what you think and I'd be happy to make changes if you'd like.

Reply:
> I figured it made the most sense to stick to the [MIME type](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types) to imply the connection to the file input's accept attribute.

FWIW sticking to the MIME type is a good call in my opinion. It will be easier for us to convey to the consumers how this option field should be populated - Furthermore people familiar with the HTML spec of `input type="file"` will have pretty much a zero learning curve. 🙂